test/integration/shoots/operations/clusterautoscaler.go

```diff
@@ -165,6 +166,125 @@ var _ = ginkgo.Describe("Shoot clusterautoscaler testing", func() {
 		framework.ExpectNoError(err)
 	}, testTimeout)
 
+	f.Beta().Serial().CIt("should autoscale a single worker group to/from zero", func(ctx context.Context) {
+		var (
+			shoot = f.Shoot
+
+			origClusterAutoscalerConfig = shoot.Spec.Kubernetes.ClusterAutoscaler.DeepCopy()
+			workerPoolName              = shoot.Spec.Provider.Workers[0].Name
+			testWorkerPoolName          = "ca-test"
+			origWorkers                 = shoot.Spec.Provider.Workers
+		)
+		if shoot.Spec.Provider.Type != "aws" && shoot.Spec.Provider.Type != "azure" {
+			ginkgo.Skip("not applicable")
+		}
+
+		// Create a dedicated worker-pool for cluster autoscaler.
+		testWorkerPool := shoot.Spec.Provider.Workers[0]
+		testWorkerPool.Name = testWorkerPoolName
+		testWorkerPool.Minimum = 0
+		testWorkerPool.Maximum = 2
+		testWorkerPool.Labels = map[string]string{"worker.gardener.cloud/pool": testWorkerPoolName}
```

Comment:
Isn't that label added by the kubelet already during node registration? Or why is there a need to add that label explicitly?

Reply:
This is actually related to the scale-to-zero feature. This label is for future node-objects, which will be created under this worker/machine-deployment. 
Cluster-autoscaler reads this from MachineDeployment and decides if reserve-excess-capacity pods[with label affinity]  will be able to schedule on the future nodes under this Machinedeployment. 
I hope that's understandable, essentially it's dependent on how autoscaler works.

Reply:
If I understand it correctly, this label is required because the cluster-autoscaler reads it from the  machine deployment? 
Is Gardener creating these labels during the Worker creation or who is setting these labels on the deployment? If yes, is Gardener always setting thee labels because they are required for the autoscaler feature? 
If that is the case, we can get rid of the[ kubelet flag to not have redundancy.](https://github.com/gardener/gardener/blob/51634b550b0819af24d2dd1a08d5a15288b48d46/charts/seed-operatingsystemconfig/original/templates/kubelet/_kubelet.flags#L23)

Reply:
Got your point.
The `worker.gardener.cloud/pool` label is added by the kubelet itself (via flags configured in the systemd unit), but the cluster-autoscaler needs to know which labels a node in a specific pool will have, so it will look at the MachineDeployment, right?
So if I understand correctly what you are saying, an enduser who wants to use the scale-from-zero feature would need to explicitly add this label to the worker pool in his Shoot spec?
If that's the case, I would propose to enhance the Worker actuator to always add the labels, a pool will have, to the MachineDeployment.

Reply:
>So if I understand correctly what you are saying, an end-user who wants to use the scale-from-zero feature would need to explicitly add this label to the worker pool in his Shoot spec?

Not really. I added this label only because I wanted the reserve-excess-capacity pods to be scheduled on this worker-pool only. In general, end-users don't need to add this label in shoot-spec unless their workload has a specific affinity towards the worker-pool. 
- Technically it could be any `test-label` and not necessarily worker pool-specific. 

>If that's the case, I would propose to enhance the Worker actuator to always add the labels, a pool will have, to the MachineDeployment.

So I think worker-actuator doesn't need any changes.

Reply:
> I added this label only because I wanted the reserve-excess-capacity pods to be scheduled on this worker-pool only

Sure. And what if the enduser wants to do the same and run some portion of his workload on a dedicated worker pool, which should be scaled down to / up from zero? Then they would need to explicitly add this label to the worker pool in his Shoot spec, correct?

Reply:
>Then they would need to explicitly add this label to the worker pool in his Shoot spec, correct? 

Yes, that's a very valid point, and possibly a gap also I think.

Reply:
Ok, then WDYT about enhancing the worker actuator to add the default pool labels to the MachineDeployment(s)? This would solve this problem, right?

Does mcm add then add these labels from MachineDeployments to the Node objects then? If so, then we should indeed avoid duplication, as @danielfoehrKn suggested (see https://github.com/gardener/gardener/pull/2160#discussion_r405413204).

Reply:
>Does mcm add then add these labels from MachineDeployments to the Node objects then? 

Yes. I generally agree to set the worker-pool label via MachineDeployment, in that case, autoscaler would be able to read them and workload having an affinity to this label can basically trigger the scale-up of zeroed worker pools.

Though, I would also expect end-users to completely control the labels, if their workload is dependent on them via affinity.

Reply:
Perfect, then let's adapt this. But anyways not in this PR, I will open an issue for that.

> Though, I would also expect end-users to completely control the labels, if their workload is dependent on them via affinity.

Sure, of course, we should add the default labels (like `worker.gardener.cloud/pool` and so on) and any additionally specified worker pool labels from the Shoot spec to the MachineDeployment.

@rfranzke Should we adapt the worker contract and add that the worker controller is responsible for adding the node labels to avoid duplication? Or is there a specific reason, why we add the labels through kubelet flags?

Reply:
Yes, sounds good. But I suggest to do it like @danielfoehrKn did it for the container runtime labels - let Gardener create the `Worker` extension object with the respective labels. Then we don't have to revendor the generic worker actuator in all extensions and get it out of the box. Also, it makes more sense to put there as Gardener can control this label centrally - it's not necessary that every worker extension does it on its own.

https://github.com/gardener/gardener/pull/2128/files#diff-873d79d4ef156d5ab5a81e3dce9a0a41 - @tim-ebert you want to take this?

Reply:
Yep, that's exactly what I meant. I can take this 👍

Reply:
Is this a prerequisite for this PR?

Reply:
No, we can go ahead with this one in the meantime.

Reply:
There you go: https://github.com/gardener/gardener/pull/2166

Reply:
I tested both clusterautoscaler integration tests with #2166, and they worked out perfectly fine. It also worked without setting the worker pool label explicitly in your test. So we can go ahead and remove it :)

Reply:
Sure, I removed the label from the worker pool, It would be nice if you could have a final check.